### ISSUE TITLE
Swagger expand add invalid authorizationUrl key in securityDefinitions

### ISSUE THREAD

**Issue by patrick-hahn-sit** [go-swagger/go-swagger#2506]
## Problem statement

When using `swagger expand` on a Swagger document with a OAuth2 flow: application security definition, `expand` adds an invalid empty string key authorizationUrl to the security definition.

The authorizationUrl key is only valid when used with flow: implicit or accessCode, not with application.

## Swagger specification
```yaml
swagger: "2.0"
securityDefinitions:
  OAuth2:
    type: oauth2
    flow: application
    tokenUrl: https://example.com/oauth/token
```

## Steps to reproduce
1. Run `swagger expand test.yaml`
2. The output contains an empty string key authorizationUrl in the key
```yaml
swagger: "2.0"
paths: null
securityDefinitions:
  OAuth2:
    type: oauth2
    flow: application
    authorizationUrl: ""
    tokenUrl: https://example.com/oauth/token
```

## Environment
swagger version: 
```
version: v0.26.0
commit: 640fa2f19f215cda0b815a4bb9a26aae2bd477d2
```
go version: 
`go version go1.15.3 darwin/amd64`
 
OS:  Mac OSX 10.14.5

**Comment by elipavlov** [comment]
It seems that behaviour reproduced in the v0.26.1.

@fredbi 

Steps:
```shell
# enter docker container with go-swagger:v0.26.1
docker run --rm --name sw -it -v $(pwd):/src --entrypoint /bin/ash quay.io/goswagger/swagger:v0.26.1

# see original file
/go # cat /src/sw3.yml
swagger: "2.0"
securityDefinitions:
  OAuth2:
    type: oauth2
    flow: application
    tokenUrl: https://example.com/oauth/token

# go-swagger result
/go # swagger expand --format yaml /src/sw3.yml
swagger: "2.0"
paths: null
securityDefinitions:
  OAuth2:
    type: oauth2
    flow: application
    authorizationUrl: ""
    tokenUrl: https://example.com/oauth/token
```

**Comment by jokaio** [comment]
@elipavlov it was fixed with v0.20.3 of the go-openapi/spec which includes fredbi's fix https://github.com/go-openapi/spec/blob/v0.20.3/security_scheme.go#L128 
The go-swagger version 0.26.1 which you mentioned still uses v0.20.2 of the go-openapi/spec, but the current master branch already depends on the newer v0.20.3 so I would assume that this issue here will be fixed as part of the next release.

**Comment by elipavlov** [comment]
@jokaio thanks for the explanation.

Sorry for the false alert.
